infra/Pos/Slotting/Ntp.hs

```diff
@@ -94,12 +108,29 @@ type SlottingConstraint m =
 
 instance SlottingConstraint m =>
          MonadSlots (NtpSlotting m) where
-    getCurrentSlot = ntpGetCurrentSlot
-    getCurrentSlotBlocking = ntpGetCurrentSlotBlocking
-    getCurrentSlotInaccurate = ntpGetCurrentSlotInaccurate
-    currentTimeSlotting = ntpCurrentTime
+    getCurrentSlot =
+        ifNtpUsed (lift simpleGetCurrentSlot)
+                  (local' ntpGetCurrentSlot)
+    getCurrentSlotBlocking =
+        ifNtpUsed (lift simpleGetCurrentSlotBlocking)
+                  (local' ntpGetCurrentSlotBlocking)
+    getCurrentSlotInaccurate =
+        ifNtpUsed (lift simpleGetCurrentSlotInaccurate)
+                  (local' ntpGetCurrentSlotInaccurate)
+    currentTimeSlotting =
+        ifNtpUsed (lift simpleCurrentTimeSlotting)
+                  (local' ntpCurrentTime)
     slottingWorkers = [ntpSyncWorker]
 
+boolM :: Monad m => m a -> m a -> m Bool -> m a
+boolM f t b = b >>= bool f t
+
+local' :: Monad m => RealNtpSlotting m a -> NtpSlotting m a
+local' (TaggedTrans tr) = TaggedTrans $ withReaderT snd tr
+
+ifNtpUsed :: Monad m => NtpSlotting m a -> NtpSlotting m a -> NtpSlotting m a
+ifNtpUsed f t = Ether.asks @(Bool, NtpSlottingVar) fst >>= bool f t
```

Comment:
I guess you're once again making things unobvious by using "if" prefix and having false-first order of arguments. Might be confusing. Consider switching `f` and `t`, renaming function or adding a warning comment.